### ISSUE TITLE
[Downgrade] - Returning self no longer has proper type annotations

### ISSUE THREAD

**Issue by ianhco** [microsoft/pylance-release#1927]
Today I noticed something is wrong with the annotation of codes, lot of codes and variables are marked as `Any` (white color), I have iterated the possible cause of this can it was due to `return self` in an `async` method defined in parent classes. It does not annotate properly `self` when it is extended, it keeps annotating `self` as the parent class. See image:

![image](https://user-images.githubusercontent.com/58193352/136697866-da44fc99-acc4-4501-a8a4-f150be0c9eb1.png)

Note: this only happens to `async` methods, non-async methods works as expected.

## Environment data

Language server version: 2021.10.0
Pylance version: v2021.10.0
Python version: 3.8.7
OS: win10

## Expected behaviour

Variable `o` in image is annotated as `B` instead of `~A`

## Actual behaviour

Variable `o` in image is annotated as `~A`

## Impact potential

it basically disables the ability of autocompletion after using a parent defined async method returning self, because it is annotated as the parent class, it does not have access to extended class attributes.

## Code snippet of the image

```python
class A:

    async def get(self):
        return self


class B(A):
    pass


async def run():
    o = await B().get()
    o
```

## Temporary Fix

**EDIT: Downgrading Pylance to version v2021.9.4 fixes this aswell. Peraphs the easiest fix**

```python
from typing import TypeVar

Self = TypeVar('Self')

class A:

    async def get(self: Self) -> Self:
        return self

```

This does not justify as a "expected behavior", it functions with normal python methods but not coroutine methods

**Comment by paaksing** [comment]
@erictraut Thanks, may I know when will next release be ?

**Comment by bschnurr** [comment]
This issue has been fixed in version 2021.10.1, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2021101-14-october-2021
